File: 938-range-sum-of-bst/938-range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0;
    void helper(TreeNode root, int L, int R){
         if(root == null){
            return;
        }
        if(root.val <= R && root.val >= L){
            sum += root.val;
            helper(root.left, L, R);
            helper(root.right, L, R);
        }else if(root.val < L){
            helper(root.right, L, R); //no need to go left since the left branch must be smaller than L
        }else{ //root great than R
            helper(root.left, L, R); //no need to go right since the right branch must be larger than R
        }
        
    }
    public int rangeSumBST(TreeNode root, int low, int high) {
        
        helper(root,low,high);
        
        return sum;
        
        
    }
}

File: 938-range-sum-of-bst/938-range-sum-of-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void sumroot(TreeNode* root, int low, int high, int &sum){
        if(root==NULL){
            return;
        }
        if(root->val>=low&&root->val<=high){
            sum+=root->val;
            
        }
         sumroot(root->left,low,high,sum);
          sumroot(root->right,low,high,sum);
    }
    int rangeSumBST(TreeNode* root, int low, int high) {
        int sum=0;
        sumroot(root,low,high,sum);
        return sum;
        
        
    }
};